labgrid/remote/coordinator.py

```diff
@@ -672,6 +677,27 @@ async def release_place(self, name, details=None):
         self.schedule_reservations()
         return True
 
+    @locked
+    async def release_place_from(self, name, acquired, details=None):
+        try:
+            place = self.places[name]
+        except KeyError:
+            return False
+        if not place.acquired:
+            return True
+        if place.acquired != acquired:
+            return True
```

Comment:
Could you explain why you return `True` in these cases?

Reply:
The logic is that the operation returns `True` if the supplied user no longer owns the place, and `False` if there was an error. The user could no longer own the place either because it's not acquired, another user has it acquired, or it was released for the requested user. Either way, the end result is the same, so `True` is returned. Put another way, returning `False` in these cases isn't helpful because we don't care if the place is acquired, or acquired by another user; as long as it's not acquired by the provided user when this function returns, we consider that success.

Reply:
Hm, that differs from the "normal" release logic, right? Releasing an already released place is considered an error.

Reply:
I don't have a strong opinion regarding this, I just want to make sure these things follow a consistent logic.

Reply:
That is correct.... however, based on the intention of this API, returning False in that case makes the return value useless for scripting purposes; you would basically have to ignore the return because it's impossible to predict if it will return True or False, and impossible to know if False is an error, or just a race releasing the place.

Reply:
Yes, agreed. I'm okay with this then.

Reply:
Perhaps a comment here describing the reasoning would be useful to avoid confusion in the future. Or even better a short section in the docs?